components/AddCashExpenseModal.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState } from "react";
+
+// ICONS
+import AttachMoneyIcon from "@mui/icons-material/AttachMoney";
+import CloseIcon from "@mui/icons-material/Close";
+
+// COMPONENTS
+import IconButton from "@mui/material/IconButton";
+import { Tooltip, Modal, Box } from "@mui/material";
+import TextField from "@mui/material/TextField";
+
+const style = {
+  position: "absolute",
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  width: 280,
+  bgcolor: "background.paper",
+  boxShadow: 24,
+  borderRadius: 1,
+  overflow: "hidden",
+};
+
+interface Expense {
+  amount: string; // Or movement
+  place: string;
+  description: string;
+  origin: string;
+  created_at: string;
+  user_id: number;
+  goal_id: number;
+  category_id: number;
+  // + id
+}
+
+const AddCashExpenseModal = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [expensesArray, setExpensesArray] = useState<Expense[]>([]);
+  const [expenseForm, setExpenseForm] = useState({
+    amount: "",
+    place: "",
+    description: "",
+    origin: "",
+    /*created_at: "",
+    user_id: 0,
+    goal_id: 0,
+    category_id: 0,*/
+  });
+
+  const openCashModal = () => {
+    setIsModalOpen(!isModalOpen);
+    console.log(isModalOpen);
+  };
+
+  const handleClose = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleAdd = () => {
+    if (expenseForm.amount !== "" && expenseForm.place !== "") {
+      const newExpense = {
+        ...expenseForm,
+        created_at: new Date().toISOString(),
+        user_id: 1,
+        goal_id: 1,
+        category_id: 1,
+      };
+
+      setExpensesArray((prevExpenses) => [...prevExpenses, newExpense]);
+
+      setExpenseForm({
+        amount: "",
+        place: "",
+        description: "",
+        origin: "",
+      });
+
+      console.log(expensesArray);
+    } else {
+      window.alert("Something is missing!");
+    }
+  };
+
+  const handleFormChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setExpenseForm((prev) => ({ ...prev, [e.target.name]: e.target.value }));
+  };
+
+  return (
+    <div>
+      <Tooltip arrow title={"Add cash expense"}>
+        <IconButton onClick={openCashModal} sx={{ mr: "0.5rem" }}>
+          <AttachMoneyIcon />
+        </IconButton>
+      </Tooltip>
+      <Modal open={isModalOpen} onClose={handleClose}>
+        <Box sx={style}>
```

Comment:
Is it possible to use TailwindCSS here? Not a problem, but it would be an improvement for maintainability.

Reply:
Good for this PR!